synapse/replication/tcp/client.py

```diff
@@ -197,9 +260,16 @@ async def on_position(self, stream_name: str, instance_name: str, token: int):
         # may be streaming.
         self.notifier.notify_replication()
 
+        await self.on_rdata(stream_name, instance_name, token, [])
```

Comment:
can we get rid of the call to `self.store.process_replication_rows` above, given that is also called by `on_rdata` ?

Reply:
Good point

Reply:
Done. I've moved the `self.notifier.notify_replication()` after the `on_rdata(..)` call in case anything listening was requiring the store to have processed the replication.